### PR TITLE
Fixed zephir_call_function_opt (set call->This)

### PR DESCRIPTION
**Pull request by sergeyklay**
@sjinks @dreamsxin Could you please tale a look

-------

Refs: https://github.com/phalcon/zephir/issues/1392

### CODE REVIEW

**Comment by sjinks** [comment]
Looks good to me. Well done!

**Comment by sergeyklay** [comment]
Thanks to @ovr

**Comment by dreamsxin** [comment]
Good job

**Comment by sergeyklay** [comment]
@dreamsxin But tests now failed. I've rollback in this repo https://github.com/sergeyklay/issue1394

File: kernels/ZendEngine3/extended/fcall.c [hunk 1]
```diff
@@ -1197,9 +1197,12 @@ int zephir_call_function_opt(zend_fcall_info *fci, zend_fcall_info_cache *fci_ca
 
 #if PHP_VERSION_ID >= 70100
 	//EG(fake_scope) = calling_scope;
-        if(fci_cache->object) {
-            Z_OBJ(call->This) = fci_cache->object;
-        }
+	if (fci_cache->object) {
+		Z_OBJ(call->This) = fci_cache->object;
+	} else {
+		Z_OBJ(call->This) = fci->object;
```
Comment (sergeyklay) [inline kernels/ZendEngine3/extended/fcall.c:1203]: @dreamsxin 
```gdb
(gdb) print fci.object.ce
Cannot access memory at address 0x10

(gdb) print fci.object.properties
Cannot access memory at address 0x20

(gdb) print fci.object.properties_table
Cannot access memory at address 0x28

(gdb) print fci.object
$8 = (zend_object *) 0x0

(gdb) print fci.size
$11 = 56

(gdb) print fci_cache.object.ce
$15 = (zend_class_entry *) 0x7ffff3803240
```

So looks like `fci.object` empty here

Comment (sergeyklay) [reply kernels/ZendEngine3/extended/fcall.c:1203]: Code before 
```c
#if PHP_VERSION_ID >= 70100
	fci->object = (func->common.fn_flags & ZEND_ACC_STATIC) ? NULL : fci_cache->object;
	call = zend_vm_stack_push_call_frame(ZEND_CALL_TOP_FUNCTION, func, fci->param_count, fci_cache->called_scope, fci->object);
#else
```

Lets see
```gdb
(gdb) print func.common.fn_flags
$18 = 513

(gdb) print ZEND_ACC_STATIC
$19 = 1

(gdb) print 513 & 1
$21 = 1
```

So `fci->object` is NULL

Comment (dreamsxin) [reply kernels/ZendEngine3/extended/fcall.c:1203]: Fllow `zend_get_called_scope`,  If there is no `object` then use `ce` set to `This`
```c
ZEND_API zend_class_entry *zend_get_called_scope(zend_execute_data *ex) /* {{{ */
{
	while (ex) {
		if (Z_TYPE(ex->This) == IS_OBJECT) {
			return Z_OBJCE(ex->This);
		} else if (Z_CE(ex->This)) {
			return Z_CE(ex->This);
		} else if (ex->func) {
			if (ex->func->type != ZEND_INTERNAL_FUNCTION || ex->func->common.scope) {
				return NULL;
			}
		}
		ex = ex->prev_execute_data;
	}
	return NULL;
}
``

Comment (dreamsxin) [reply kernels/ZendEngine3/extended/fcall.c:1203]:
```c
static zend_always_inline void zend_vm_init_call_frame(zend_execute_data *call, uint32_t call_info, zend_function *func, uint32_t num_args, zend_class_entry *called_scope, zend_object *object)
{
	call->func = func;
	if (object) {
		Z_OBJ(call->This) = object;
		ZEND_SET_CALL_INFO(call, 1, call_info);
	} else {
		Z_CE(call->This) = called_scope;
		ZEND_SET_CALL_INFO(call, 0, call_info);
	}
	ZEND_CALL_NUM_ARGS(call) = num_args;
}
```

**Pull request merged**